File: Data Structures & Algorithms/evaluate-reverse-polish-notation/submission-1.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:

        operStack = []

        while len(tokens) != 0:
            operStack.append(tokens.pop())

        total = float(operStack.pop())

        while len(operStack) != 0:
            number = float(operStack.pop())
            operation = operStack.pop()
            print (total, number, operation)

            if operation == "+":
                total += number
            if operation == "-":
                total == number
            if operation == "*":
                total *= number
            if operation == "/":
                total /= number

        return int(total)
            


        

File: Data Structures & Algorithms/evaluate-reverse-polish-notation/submission-1.java
class Solution {
    public int evalRPN(String[] tokens) {

        Deque<Integer> numStack = new ArrayDeque<>();

        for (String token : tokens) {

            if (token.equals("+") || token.equals("-") || token.equals("*") || token.equals("/")) {
                int num1 = numStack.pop();
                int num2 = numStack.pop();

                if (token.equals("+")) {
                    numStack.push(num2 + num1);
                } 
                if (token.equals("-")) {
                    numStack.push(num2 - num1);
                }
                if (token.equals("*")) {
                    numStack.push(num2 * num1);
                } 
                if (token.equals("/")) {
                    numStack.push(num2 / num1);                
                }

            } else {
                numStack.push(Integer.parseInt(token));
            }
        }

        return numStack.pop();
    }
}